src/oom/oomd.c

```diff
@@ -131,8 +133,16 @@ static int run(int argc, char *argv[]) {
 
         /* Do some basic requirement checks for running systemd-oomd. It's not exhaustive as some of the other
          * requirements do not have a reliable means to check for in code. */
-        if (access("/proc/swaps", F_OK) < 0)
-                return log_error_errno(errno, "Swap not enabled: %m");
+
+        r = get_proc_field("/proc/meminfo", "SwapTotal", WHITESPACE, &swap);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get SwapTotal from /proc/meminfo: %m");
+
+        r = safe_atollu(swap, &s);
+        if (r < 0)
+                return log_error_errno(r, "Failed to parse SwapTotal from /proc/meminfo: %s: %m", swap);
+        if (s == 0)
+                return log_error_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Requires swap to operate");
```

Comment:
hmm, on a kernel compiled without swap support, how will the error message look like now? i.e. is SwapTotal in /proc/meminfo even then and hardcoded to 0?

Reply:
(if that's the case, maybe add a brief comment clarifying that the field exists even on swap-disabled kernels)